### PR TITLE
Add metric to Solace Scaler

### LINKED ISSUE #4665: Add new "messageReceiveRateTarget" metric to Solace Scaler
### Proposal

Add a new **messageReceiveRateTarget** metric to Solace Scaler. This metric will allow KEDA to scale workloads based on the rate of message delivery to Solace PubSub+ queues. 

### Use-Case

This enhancement will permit KEDA to maintain workloads at a level desired measured by the rate of message ingress to queues. Thus reducing the tendency of flapping (creation and destruction of pods) as queue backlogs are cleared.
Currently available metrics of **messageCountTarget** and **messageSpoolUseTarget** allow KEDA to scale workloads based upon existing backlogs. The new metric allows KEDA to scale workloads based upon the rate of message ingress. This feature has two benefits:
- Provides a mechanism to maintain performant applications when existing backlogs are small
- Provides a method to reduce pod "flapping," the continuous creation and destruction of pods, as KEDA maintains workload replicas in concert with observed demand in the form of queue backlogs.

### Is this a feature you are interested in implementing yourself?

Yes

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by dennis-brinley**
Added new metric: **messageReceiveRateTarget** to Solace Scaler

### Checklist

- [X] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [X] Tests have been added
- [X] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [X] A PR is opened to update the documentation on https://github.com/kedacore/keda-docs/pull/1150
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #4665

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 - **Security:** Enable secret scanning in GitHub repo
 - **RabbitMQ Scaler**: Add support for `unsafeSsl` in trigger metadata ([#4448](https://github.com/kedacore/keda/issues/4448))
 - **Prometheus Metrics**: Add new metric with KEDA build info ([#4647](https://github.com/kedacore/keda/issues/4647))
+- **Solace Scaler**: Add new "messageReceiveRateTarget" metric to Solace Scaler ([#4665](https://github.com/kedacore/keda/issues/4665))
```
Comment (zroubalik) [inline CHANGELOG.md:65]:
```suggestion
- **Solace Scaler**: Add new `messageReceiveRateTarget` metric to Solace Scaler ([#4665](https://github.com/kedacore/keda/issues/4665))
```

Just a nit.

**Review by zroubalik — COMMENTED** [review]
Lookin good, please fix the e2e test - see Static Checks failure, you can ignore the "Welcome PR" failure.

File: CHANGELOG.md (hunk 1 above)
Comment (dennis-brinley) [reply CHANGELOG.md:65]: Updated

**Comment by dennis-brinley** [comment]
@zroubalik - there were problems with the helm chart references (the chart url context changed) - which I didn't detect because I was skipping that part of the test. I updated the code and ran the test successfully on my machine. Should be good to go.

**Comment by zroubalik** [comment]
I am not possible to merge the PR, could you please rebase your branch? Thanks!

**Review by zroubalik — COMMENTED** [review]
@dennis-brinley I am not possible to merge the PR, could you please rebase your branch? Thanks!

**Comment by dennis-brinley** [comment]
merge conflict should be resolved

**Review by zroubalik — COMMENTED** [review]
@dennis-brinley I can see that your rebase unfortunately included additional commits in your PR. You can see that there are some unrelevant changes. Could you please fix that?

**Comment by dennis-brinley** [comment]
chopped and re-committed

**Comment by dennis-brinley** [comment]
Tests failed on predictkube scaler. I don't think there is anything I can do about that directly.

```
2023/06/20 20:54:39 🚀 starting mock grpc server. On host 0.0.0.0, with port: 40761
    predictkube_scaler_test.go:225: get: {{10130000 -3} {<nil>}  DecimalSI}, want: {{10130 0} {<nil>}  DecimalSI}, predictMetric: 10130
    predictkube_scaler_test.go:221: 
        	Error Trace:	/__w/keda/keda/pkg/scalers/predictkube_scaler_test.go:221
        	Error:      	Received unexpected error:
        	            	Post "http://demo.robustperception.io:9090/api/v1/query_range": read tcp4 172.19.0.2:60472->157.245.26.111:9090: i/o timeout
        	Test:       	TestPredictKubeGetMetrics
    predictkube_scaler_test.go:222: 
        	Error Trace:	/__w/keda/keda/pkg/scalers/predictkube_scaler_test.go:222
        	Error:      	Not equal: 
        	            	expected: 0
        	            	actual  : 1
        	Test:       	TestPredictKubeGetMetrics
--- FAIL: TestPredictKubeGetMetrics (20.08s)
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 11020 [running]:
testing.tRunner.func1.2({0x2495f80, 0x400363a060})
	/usr/local/go/src/testing/testing.go:1526 +0x1c8
testing.tRunner.func1()
	/usr/local/go/src/testing/testing.go:1529 +0x364
panic({0x2495f80, 0x400363a060})
	/usr/local/go/src/runtime/panic.go:884 +0x1f4
github.com/kedacore/keda/v2/pkg/scalers.TestPredictKubeGetMetrics(0x400309c680)
	/__w/keda/keda/pkg/scalers/predictkube_scaler_test.go:223 +0x3b4
testing.tRunner(0x400309c680, 0x27d12f8)
	/usr/local/go/src/testing/testing.go:1576 +0x104
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1629 +0x370
FAIL	github.com/kedacore/keda/v2/pkg/scalers	28.280s
```

**Pull request merged**